sw/device/lib/testing/usb_testutils_streams.c

```diff
@@ -0,0 +1,426 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// USB streaming data test
+//
+// This test requires interaction with the USB DPI model or a test application
+// on the USB host. The test initializes the USB device and configures a set of
+// endpoints for data streaming using bulk transfers.
+//
+// The DPI model mimicks a USB host. After device initialization, it detects
+// the assertion of the pullup and first assigns an address to the device.
+// For this test it will then repeatedly fetch data via IN requests to
+// each stream and propagate that data to the corresponding OUT endpoints.
+//
+// The data itself is pseudo-randomly generated by the sender and,
+// independently, by the receiving code to check that the data has been
+// propagated unmodified and without data loss, corruption, replication etc.
+
+//#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/usb_testutils_streams.h"
+
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/usb_testutils_diags.h"
+
+/**
+ * Read method to be employed
+ */
+static const enum {
+  kReadMethodNone = 0u,  // Just discard the data; do not read it from usbdev
+  kReadMethodFaster,     // Faster implementation
+  kReadMethodStandard    // Use standard dif_usbdev_buffer_read() function
+} read_method = kReadMethodFaster;
```

Comment:
Sorry for don't noticing this before, but in the first commit this enum is being defined in the file `sw/device/tests/usbdev_stream_test.c`, so could you move this definition to a common place? Maybe to the `usb_testutils.h`? I fine if do it in a follow-up PR.